lib/utilities/retrieveData.js

```diff
@@ -60,7 +60,17 @@ export default function retrieveData(locations, retrieveDataFn, response, log) {
                 log.error('error piping data from source');
                 return _next(err);
             });
-            return readable.pipe(response, { end: false });
+            readable.on('data', chunk => {
```

Comment:
Can you give a short hint on why this solves the memory issue in the PR?

Reply:
It's response.write method which solves the issue. I don't have a technical argument as to why, because stream.pipe essentially does the same.

Reply:
ok, good to know